.github/scripts/check-via-duplication.py

```
@@ -0,0 +1,107 @@
+    if not condition: raise ValueError(message)
+
+
+def canonical(scope, core, verifier):
+    require(scope in ROOTS, f"unknown scope {scope!r}"); paths = tuple(PurePosixPath(path) for path in (core, verifier)); roots = tuple(PurePosixPath(path) for path in ROOTS[scope])
+    require(all(path.as_posix() == raw and not path.is_absolute() and ".." not in path.parts and path.suffix == ".rs" for path, raw in zip(paths, (core, verifier))), "policy paths must be canonical repository-relative Rust paths")
+    require(all(path.is_relative_to(root) for path, root in zip(paths, roots)), "policy paths do not match their declared scope")
+    return scope, core, verifier
+
+
+def rows(path, header, width):
+    lines = Path(path).read_text().splitlines(); require(lines and lines[0] == header and all(lines[1:]), f"invalid header or blank row in {path}")
```

Ruff (0.15.21)

[error] Multiple statements on one line (semicolon)

(E702)